### DISCUSSION TITLE
Can't add a database when using OIDC (Keycloak)

### DISCUSSION THREAD

**Discussion by manodupont** [apache/superset#18362] [Q&A / Help]
I connected superset with Keycloak.

I pretty much followed some forum here and there and finally got this working.

The problem is : Why did I lose the possibility to "Add a database" or a "dataset". I dont have that menu anymore.
<img width="1536" alt="Screen Shot 2021-12-20 at 11 27 38" src="https://user-images.githubusercontent.com/7568033/146752819-46da7e5c-fd4e-4e1f-99f5-dd6320978f94.png">

### Expected results

Can add a database

### Actual results

I cannot add a database

### Environment

docker image `apache/superset:latest-dev`

**Comment by srinify** [comment]
This usually happens when you're logged into a `role` in Superset that doesn't have access / permissions to add database. I've never used Keycloak but this is the surface area that I would attempt to debug. What roles have you created in Superset, which one are you logged in as right now, etc.

**Comment by manodupont** [comment]
Well yes i do understand. I  think this is the probleme cause even though I force the user logged to be "Admin" it still shows as "Gamma" when I am logged in, in the user profile.

So, i mean, any information on how role works and how I can map roles into my SecurityManager would be appreciated.

Right now my superset_config.py looks like this : 
```
...
from keycloack_security_manager  import  OIDCSecurityManager
from flask_appbuilder.security.manager import AUTH_OID, AUTH_REMOTE_USER, AUTH_DB, AUTH_LDAP, AUTH_OAUTH
import os

#---------------------------KEYCLOACK ----------------------------

curr  =  os.path.abspath(os.getcwd())
AUTH_TYPE = AUTH_OID
SECRET_KEY: 'SomethingNotEntirelySecret'
OIDC_CLIENT_SECRETS =  curr + '/docker/pythonpath_dev/client_secrets.json'
OIDC_ID_TOKEN_COOKIE_SECURE = False
OIDC_REQUIRE_VERIFIED_EMAIL = False
OIDC_OPENID_REALM: 'portal-cockpit'
OIDC_INTROSPECTION_AUTH_METHOD: 'client_secret_post'
CUSTOM_SECURITY_MANAGER = OIDCSecurityManager
AUTH_USER_REGISTRATION = True
AUTH_USER_REGISTRATION_ROLE = 'Admin'

#--------------------------------------------------------------
...
```

So i thought that `AUTH_USER_REGISTRATION_ROLE = 'Admin'` would set anything to "Admin" but it doesn't seem to work.

**Comment by jhult** [comment]
@manodupont, copy/paste this class: `OIDCSecurityManager` (`from keycloack_security_manager`). Also, FYI, `keycloack_security_manager` has a typo.
